lib/auth/auth.go

```diff
@@ -423,10 +435,116 @@ func (a *Server) runPeriodicOperations() {
 			}
 			// Update prometheus gauge
 			heartbeatsMissedByAuth.Set(float64(missedKeepAliveCount))
+		case <-promTicker.C:
+			a.versionMetrics()
 		}
 	}
 }
 
+// versionMetrics leverages the cache to report all agents/proxies/auth servers connected to the
+// teleport cluster via prometheus metrics
+func (a *Server) versionMetrics() {
```

Comment:
Maybe call this `updatePrometheusMetrics` or something?

Reply:
I'm hesitant to change this as there may be other metric related functions added in the future.

Reply:
It's unexported, so we can change the name whenever we want.

We should choose a name that most accurately reflects what it does _today_, not what it may do in the future.

`versionMetrics` is unclear to me. Is "version" a verb in this case?